core/src/main/java/hudson/security/SecurityRealm.java

```diff
@@ -290,13 +290,32 @@ public void doLogout(StaplerRequest req, StaplerResponse rsp) throws IOException
         Authentication auth = SecurityContextHolder.getContext().getAuthentication();
         SecurityContextHolder.clearContext();
 
-        // reset remember-me cookie
-        Cookie cookie = new Cookie(ACEGI_SECURITY_HASHED_REMEMBER_ME_COOKIE_KEY,"");
-        cookie.setMaxAge(0);
-        cookie.setSecure(req.isSecure());
-        cookie.setHttpOnly(true);
-        cookie.setPath(req.getContextPath().length()>0 ? req.getContextPath() : "/");
-        rsp.addCookie(cookie);
+        String contextPath = req.getContextPath().length() > 0 ? req.getContextPath() : "/";
+        {
```

Comment:
Why the block?

Reply:
I used it to distinguish the two of users of `Cookie cookie`, but I guess that isn't really necessary.

Reply:
I take that back, it is necessary. Java won't allow me to not type my cookie in the for loop, and it won't let me declare the same object both in the function scope and in the for loop scope.

That alternative is that I define two differently named cookie variables. If you have a preference between the two, I'm open to suggestions. Personally, I think using scoped variables is better than naming one `cookie2` or `c` or whatever.

Reply:
I would probably name the one in the for loop on line 303 something like "currentCookie". Or maybe pull one or both parts of the behavior out into methods that describe what they're doing. That's probably my favorite.

But, this is getting to the level of personal preference and I really don't care which one you go with.